src/validation/gltf/lightsPunctual/MaxarTemporalLightTraits/MaxarTemporalLightTraitsValidator.ts

```diff
@@ -0,0 +1,190 @@
+import { JsonValidationIssues } from "../../../../issues/JsonValidationIssues";
+
+import { GltfExtensionValidationIssues } from "../../../../issues/GltfExtensionValidationIssues";
+
+/**
+ * A class for validating the `MAXAR_temporal_light_traits` extension
+ * that extends `KHR_lights_punctual` lights.
+ *
+ * @internal
+ */
+export class MaxarTemporalLightTraitsValidator {
+  /**
+   * Validates the MAXAR_temporal_light_traits extension
+   *
+   * @param path - The path for validation issues
+   * @param extension - The temporal light traits extension object
+   * @param context - The validation context
+   * @returns Whether the extension is valid
+   */
+  static validateTemporalLightTraits(
+    path: string,
+    extension: any,
+    context: ValidationContext
+  ): boolean {
+    // Make sure that the given value is an object
+    if (
+      !BasicValidator.validateObject(
+        path,
+        "MAXAR_temporal_light_traits",
+        extension,
+        context
+      )
+    ) {
+      return false;
+    }
+
+    let result = true;
+
+    // Validate the flashing property (optional)
+    const flashing = extension.flashing;
+    if (defined(flashing)) {
+      if (
+        !MaxarTemporalLightTraitsValidator.validateFlashing(
+          path + "/flashing",
+          flashing,
+          context
+        )
+      ) {
+        result = false;
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Validates a flashing trait object
+   *
+   * @param path - The path for validation issues
+   * @param flashing - The flashing object
+   * @param context - The validation context
+   * @returns Whether the flashing trait is valid
+   */
+  private static validateFlashing(
+    path: string,
+    flashing: any,
+    context: ValidationContext
+  ): boolean {
+    // Make sure that the given value is an object
+    if (!BasicValidator.validateObject(path, "flashing", flashing, context)) {
+      return false;
+    }
+
+    let result = true;
+
+    // Validate the waveform property (required)
+    const waveform = flashing.waveform;
+    const validWaveforms = ["sine", "square", "triangle"];
+    if (
+      !BasicValidator.validateEnum(
+        path + "/waveform",
+        "waveform",
+        waveform,
+        validWaveforms,
+        context
+      )
+    ) {
+      result = false;
+    }
+
+    // Validate the frequency property (required)
+    const frequency = flashing.frequency;
+    if (
+      !BasicValidator.validateNumberRange(
+        path + "/frequency",
+        "frequency",
+        frequency,
+        0.0,
+        false,
+        undefined,
+        false,
+        context
+      )
+    ) {
+      result = false;
+    }
+
+    // Validate the duty property (optional)
+    const duty = flashing.duty;
+    if (defined(duty)) {
+      if (
+        !BasicValidator.validateNumberRange(
+          path + "/duty",
+          "duty",
+          duty,
+          0.0,
+          true,
```

Comment:
A nitpick that rather refers to the _specification_: 

The descriptions at https://github.com/Maxar-Public/glTF/tree/wff1.8.0/extensions/2.0/Vendor/MAXAR_temporal_light_traits#flashing say

> ... as normalized value `(0.0, 1.0)`, only ...

The brackets _usually_ indicate whether the start and end of a range are inclusive. For example,

`[0,1]` both inclusive
`(0,1)` both exclusive
`[0, 1)` including 0, excluding 1
`(0, 1]` excluding 0, including 1

(This is also pretty consistently applied in other parts of glTF and 3D Tiles specifications)

In terms of the validation: According to the schema, the range is indeed _inclusive_, meaning that the `true` here is correct (and the range should probably be denoted as `[0,1]` in the spec text)